File: 101.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root==null) return true;
        return checker(root.left,root.right);
    }
    
    public boolean checker(TreeNode leftNode, TreeNode rightNode){
        
        if(leftNode==null && rightNode==null){
            return true;
        }
        if(leftNode==null || rightNode==null){
            return false;
        }
        
        if(leftNode.val!=rightNode.val){
            return false;
        }
                
        return checker(leftNode.left,rightNode.right) && checker(leftNode.right,rightNode.left);
        
    }
    
}

File: 101.py
# Definition for a binary tree node.
# class TreeNode:cc
#     def __init__(self, val=0, left=None, right=None):
#         self.val = vals
#         self.left = lefts
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        t1 = root
        t2 = root
        def validate(t1,t2):
            if(t1 is None and t2 is None):
                return True
            if(t1 is None or t2 is None):
                return False
            if(t1.val != t2.val):
                return False
            return validate(t1.left,t2.right) and validate(t1.right,t2.left)
        return validate(t1,t2)
        